src/backend/access/common/reloptions_gp.c

```diff
@@ -1689,9 +1689,10 @@ transformColumnEncoding(const TableAmRoutine *tam, Relation rel, List *colDefs,
 					encoding = copyObject(deflt->encoding);
 				else if (!explicitOnly)
 				{
-					if (parentenc)
+					ColumnReferenceStorageDirective *parent_col_encoding;
+					parent_col_encoding = find_crsd(d->colname, parentenc);
+					if (parent_col_encoding)
 					{
-						ColumnReferenceStorageDirective *parent_col_encoding = find_crsd(d->colname, parentenc);
 						encoding = transformStorageEncodingClause(parent_col_encoding->encoding, true);
 					}
```

Comment:
when parent_col_encoding is null, what's encoding ?

Reply:
`parent_col_encoding ` is null when adding new column for a table.